dmapiclient/base.py

```diff
@@ -120,11 +114,41 @@ def _request(self, method, url, data=None, params=None):
                 # support old .request_id attr for compatibility
                 headers[current_app.config["DM_REQUEST_ID_HEADER"]] = request.request_id
 
+        # not using CaseInsensitiveDict as our header dict initially as technically .update()'s behaviour is undefined
+        # for it, but past a certain point we want to be able to know we've resolved what our final header value is
+        # going to be for any certain header name
+        ci_headers = requests.structures.CaseInsensitiveDict(headers)
+        # just in case anyone misses the point and thinks adding anything more to `headers` will do anything beyond here
+        del headers
+
+        # determine our final outgoing span id - find the first of DM_SPAN_ID_HEADERS which is set to something truthy
+        child_span_id = next(
+            (
+                ci_headers[header_name] for header_name in (current_app.config.get("DM_SPAN_ID_HEADERS") or ())
+                if ci_headers.get(header_name)
+            ),
+            None,
+        ) if has_request_context() else None
+
+        common_log_extra = {
+            **({"childSpanId": child_span_id} if child_span_id is not None else {}),
```

Comment:
This line is failing the Python 2 flake8 check - I think we can safely take the 2.7 out of the travis.yml for this repo now?

Reply:
Though officially removing Python 2 support would probably mean a major bump. Perhaps for another time.

Reply:
People suggested I just major-bump anyway. Done now.